examples/pose-apriltag/rs-pose-apriltag.cpp

```diff
@@ -0,0 +1,296 @@
+                    std::cout << ss.str() << "camera " << print(tags.pose_in_camera[t]) << std::endl;
+                    std::cout << std::setw(ss.str().size()) << " " << "world  "<< print(tags.pose_in_world[t]) << std::endl << std::endl;
+                }
+            });
+        }
+    }
+    
+    return EXIT_SUCCESS;
+}
+catch (const rs2::error & e)
+{
+    std::cerr << "RealSense error calling " << e.get_failed_function() << "(" << e.get_failed_args() << "):\n    " << e.what() << std::endl;
+    return EXIT_FAILURE;
+}
+catch (const std::exception& e)
+{
+    std::cerr << e.what() << std::endl;
+    return EXIT_FAILURE;
+}
+
+
+matd_t* homography_compute2(double c[4][4]) {
+    double A[] =  {
+        c[0][0], c[0][1], 1,       0,       0, 0, -c[0][0]*c[0][2], -c[0][1]*c[0][2], c[0][2],
+        0,       0, 0, c[0][0], c[0][1], 1, -c[0][0]*c[0][3], -c[0][1]*c[0][3], c[0][3],
+        c[1][0], c[1][1], 1,       0,       0, 0, -c[1][0]*c[1][2], -c[1][1]*c[1][2], c[1][2],
+        0,       0, 0, c[1][0], c[1][1], 1, -c[1][0]*c[1][3], -c[1][1]*c[1][3], c[1][3],
+        c[2][0], c[2][1], 1,       0,       0, 0, -c[2][0]*c[2][2], -c[2][1]*c[2][2], c[2][2],
+        0,       0, 0, c[2][0], c[2][1], 1, -c[2][0]*c[2][3], -c[2][1]*c[2][3], c[2][3],
+        c[3][0], c[3][1], 1,       0,       0, 0, -c[3][0]*c[3][2], -c[3][1]*c[3][2], c[3][2],
+        0,       0, 0, c[3][0], c[3][1], 1, -c[3][0]*c[3][3], -c[3][1]*c[3][3], c[3][3],
+    };
+    
+    double epsilon = 1e-10;
+    
+    // Eliminate.
+    for (int col = 0; col < 8; col++) {
+        // Find best row to swap with.
+        double max_val = 0;
+        int max_val_idx = -1;
+        for (int row = col; row < 8; row++) {
+            double val = fabs(A[row*9 + col]);
+            if (val > max_val) {
+                max_val = val;
+                max_val_idx = row;
+            }
+        }
+        
+        if (max_val < epsilon) {
+            fprintf(stderr, "WRN: Matrix is singular.\n");
+        }
+        
+        // Swap to get best row.
+        if (max_val_idx != col) {
+            for (int i = col; i < 9; i++) {
+                double tmp = A[col*9 + i];
+                A[col*9 + i] = A[max_val_idx*9 + i];
+                A[max_val_idx*9 + i] = tmp;
+            }
+        }
+        
+        // Do eliminate.
+        for (int i = col + 1; i < 8; i++) {
+            double f = A[i*9 + col]/A[col*9 + col];
+            A[i*9 + col] = 0;
+            for (int j = col + 1; j < 9; j++) {
+                A[i*9 + j] -= f*A[col*9 + j];
+            }
+        }
+    }
+    
+    // Back solve.
+    for (int col = 7; col >=0; col--) {
+        double sum = 0;
+        for (int i = col + 1; i < 8; i++) {
+            sum += A[col*9 + i]*A[i*9 + 8];
+        }
+        A[col*9 + 8] = (A[col*9 + 8] - sum)/A[col*9 + col];
+    }
+   return matd_create_data(3, 3, (double[]) { A[8], A[17], A[26], A[35], A[44], A[53], A[62], A[71], 1 });
```

Comment:
g++ complains here about taking the address of a temporary array.

Reply:
updated to direct memory assignement